rfcs/0001-restructure.md

```diff
@@ -0,0 +1,271 @@
+  suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+This would result in the following order groupings in the PHP language family meta-buildpack:
+
+```toml
+[[order]] # HTTPD web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+
+[[order]] # Nginx web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+[[order]] # Built-in web server (web) & FPM (php-fpm)
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-builtin-server"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+```
+
+## Environment Variables
+
+* `php-dist` buildpack must expose the location of the `php.ini` file through
+  an environment variable available at build and launch.
+* `php-fpm` buildpack must expose the location of the `php-fpm.conf` file through
+  an environment variable available at build and launch.
+* Session handler buildpacks must use `PHP_INI_SCAN_DIR` to set the search path
```

Comment:
I think this could be obtained via `php-config`. Perhaps this section morphs from "Environment Variables" to something like "Locating PHP Resources". Then we could call out the usage of `php-config` for some info and env variables for things you can't easily get with `php-config`.

Reply:
This is just mentioning that [`PHP_INI_SCAN_DIR `](https://www.php.net/manual/en/configuration.file.php) is the right way to set a session handler. Also, I couldn't find a way to obtain scan dir with php-config.

Reply:
Sorry, I'm not sure what I was talking about here :) Ignore this comment.